docs-examples/example-java/build.gradle

```diff
@@ -0,0 +1,43 @@
+plugins {
+    id("groovy")
+    id("java")
+    id("com.github.johnrengelman.shadow") version "6.1.0"
+    id("io.micronaut.application") version "1.3.4"

```

Comment:
This is the cause of the GraalVM CI failures. There was a bug introduced in version 1.3.3 but that is fixed in version 1.4.1. But if you update the version and run `./gradlew testNativeImage` it fails differently because the Pulsar client code uses reflection in several places, so we need to find those invocations and configure native image to handle the reflection calls. I'm working on this now and will hopefully have something for that today.

Reply:
OK, I would love to learn a bit more about this one but yeah, I need to "rewrite" code from Pulsar Java client library to actually make it work then.

Reply:
It shouldn't be necessary to rewrite client code, just need to find the reflective calls and add them to the native image config files so they're expected